### PR TITLE
added Java11 base images for docker builds

### LINKED ISSUE #1647: Support for zero-config jdk 11 Docker base image
### Description
Looking to have the FMP support a Zero config JDK 11 Docker base image as it does for JDK 8. So ideally it should use the JDK 8/11 docker base image depending on whether the the compiler version of the Maven build is Java 8/11.
### Info
We ideally would want a zero-config (or close to it) option for creating a Spring Boot app using a JDK 11 Docker base image – as there is for JDK 8. I see there is a JDK 11 docker base image available as of April 2019: 
https://hub.docker.com/r/fabric8/java-centos-openjdk11-jdk
https://hub.docker.com/r/fabric8/java-centos-openjdk11-jre 
 
Also, I see there is a bug logged for JDK 11 on the maven plugin:
https://github.com/fabric8io/fabric8-maven-plugin/issues/1472
https://github.com/fabric8io/fabric8-maven-plugin/pull/1437 
https://github.com/fabric8io/fabric8-maven-plugin/projects/5 

* f-m-p version : 3.5.41
* Maven version (`mvn -v`) : 3.5.3
```

```
* If it's a feature request, what is your use case : Deploy my JDK 11 Spring Boot App in Docker

### PR DESCRIPTION
**Pull request by dev-gaur**
fixes #1647

### CODE REVIEW

File: generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java [hunk 1]
```diff
@@ -81,6 +85,40 @@ public boolean isRedHat() {
         return REDHAT_VERSION_PATTERN.matcher(version).matches();
     }
 
+    public boolean isJava11() {
```
Comment (rohanKanojia) [inline generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java]: Won't `System.getProperty("java.version")` suffice to give you jdk version?

**Comment by rohanKanojia** [comment]
@dev-gaur : Could you please resolve conflicts?

**Comment by dev-gaur** [comment]
> @dev-gaur : Could you please resolve conflicts?

working on it

File: generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java (hunk 1 above)
Comment (dev-gaur) [reply generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java]: What if the user has java11 on their Dev environment/machine but needs Java 8 on prod ?

Comment (rohanKanojia) [reply generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java]: Could you please elaborate your point? I'm not sure what you mean, properties are read from jvm and would always give correct output regardless of dev/prod environment.

File: generator/api/src/test/java/io/fabric8/maven/generator/api/FromSelectorTest.java [hunk 2]
```diff
@@ -15,18 +15,22 @@
  */
 package io.fabric8.maven.generator.api;
 
-import java.util.Map;
-
 import io.fabric8.maven.core.config.OpenShiftBuildStrategy;
-import io.fabric8.maven.core.config.RuntimeMode;
 import io.fabric8.maven.core.config.ProcessorConfig;
+import io.fabric8.maven.core.config.RuntimeMode;
```
Comment (rohanKanojia) [inline generator/api/src/test/java/io/fabric8/maven/generator/api/FromSelectorTest.java]: unused import

File: generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java [hunk 3]
```diff
@@ -34,6 +29,11 @@
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.shared.utils.StringUtils;
 
+import java.util.Collections;
```
Comment (rohanKanojia) [inline generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java]: What changes are done here?

File: generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java [hunk 4]
```diff
@@ -66,9 +71,12 @@ public String getFrom() {
     abstract protected String getS2iBuildFrom();
     abstract protected String getIstagFrom();
 
+    Plugin plugin;
```
Comment (rohanKanojia) [inline generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java]: access can be private/protected

File: generator/api/src/test/java/io/fabric8/maven/generator/api/FromSelectorTest.java (hunk 2 above)
Comment (dev-gaur) [reply generator/api/src/test/java/io/fabric8/maven/generator/api/FromSelectorTest.java]: No RuntimeMode Class is required.

File: generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java (hunk 3 above)
Comment (dev-gaur) [reply generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java]: god damn intellij!!

**Pull request merged**